### ISSUE TITLE
Is there a way to tap into the annotations saving process through this directive?

### ISSUE THREAD

**Issue by simonv3** [igorlino/angular-annotorious#4]
eg.:

```
<div annotorious-annotate on-save="saveFunction()">
</div>
```

in the controller:

```
$scope.saveFunction = function() {console.log('saving!')}
```

This would be useful for linking the annotations to a back-end, for example.

**Comment by igorlino** [comment]
Hi! sure, now there is.

I have added an example under  'src/storage-providers/'

The file is:
https://github.com/igorlino/angular-annotorious/blob/master/js/storage-providers/annotorious-local-storage.provider.js

It will save and load annotations to/from the local browser storage. This is working code.

This will give you an idea how to do it for other type of storage providers.

If you create another storage provider, that ends up generic enough, you may like to contribute it for the angular-annotorious community. So then we could place it as an example also.

**Comment by igorlino** [comment]
The way to create annotations is using the angular service interface. 
Assuming you have an image <img> element that has a src attribute.

``` javascript
var  photo = jQuery(document).find(".your-image")[0];
```

You could call:

``` javascript
annotoriousService.makeAnnotatable(photo);
```

If you want to reset/reload all annotations in the page:

``` javascript
annotoriousService.reset();
```

If you want to set the design of the annotation:

``` javascript
annotoriousService.setProperties({
            outline: '#00f',
            stroke: '#ff0000',
            fill: 'rgba(255, 0, 0, 0.3)',
            'hi_stroke': '#00ff00',
            'hi_fill': 'rgba(0, 255, 0, 0.3)'
        });
```

You define which flow/lifecycle your application follows, and you may call the reset() method a bit more often.

Saving is done transparently using the example I mentioned. No direct action required.

**Comment by igorlino** [comment]
I have updated a bit the https://github.com/igorlino/angular-annotorious/blob/master/README.md

To include this info for future users :)

If you would like to add further document that you think is useful for other developers, its very welcome.

**Comment by simonv3** [comment]
That's great, thanks for the info. Also, thanks for updating the documentation :)

**Comment by moebiusfrost** [comment]
hey guys I wanted to try the storage-provider part but it wont work because of the outdated angular lodash module

> Unable to find a suitable version for angular, please choose one by typing one of the numbers below:
    1) angular#~1.3.4 which resolved to 1.3.20 and is required by angular-annotorious#1.0.7
    2) angular#~1.4 which resolved to 1.4.14 and is required by ng-lodash#0.5.0
    3) angular#>0.9.0 which resolved to 1.6.4 and is required by angular-uuid4#0.3.1
    4) angular#>=1.3.0 which resolved to 1.6.4 and is required by ng-tags-input#3.2.0
    5) angular#^1.x which resolved to 1.6.4 and is required by angular-local-storage#0.6.0

therefore i can't get the storage-provider working. Is there a workaround for that issue?

**Comment by igorlino** [comment]
a newer version on angular lodash its compatible.
Simply declare the newer version in a resolution and bower won't complain.

https://jaketrent.com/post/bower-resolutions/
